tests/integration/mutation/create/with_default_values_test.go

```diff
@@ -0,0 +1,251 @@
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						active: Boolean @default(bool: true)
+						created: DateTime @default(dateTime: "2000-07-23T03:00:00-00:00")
+						name: String @default(string: "Bob")
+						age: Int @default(int: 40)
+						points: Float @default(float: 10)
+						metadata: JSON @default(json: "{\"one\":1}")
+						image: Blob @default(blob: "ff0099")
+					}
+				`,
+			},
+			testUtils.Request{
+				Request: `mutation {
+							create_Users(input: {age: null, active: null, name: null, points: null, created: null, metadata: null, image: null}) {
+								age
+								active
+								name
+								points
+								created
+								metadata
+								image
+							}
+						}`,
+				Results: map[string]any{
+					"create_Users": []map[string]any{
+						{
+							"age":      nil,
+							"active":   nil,
+							"name":     nil,
+							"points":   nil,
+							"created":  nil,
+							"metadata": nil,
+							"image":    nil,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestMutationCreate_WithDefaultValues_ValuesProvided_SetsValue(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Simple create mutation, with default values and values provided",
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						active: Boolean @default(bool: true)
+						created: DateTime @default(dateTime: "2000-07-23T03:00:00-00:00")
+						name: String @default(string: "Bob")
+						age: Int @default(int: 40)
+						points: Float @default(float: 10)
+						metadata: JSON @default(json: "{\"one\":1}")
+						image: Blob @default(blob: "ff0099")
+					}
+				`,
+			},
+			testUtils.Request{
+				Request: `mutation {
+							create_Users(input: {age: 50, active: false, name: "Alice", points: 5, created: "2024-06-18T01:00:00-00:00", metadata: "{\"two\":2}", image: "aabb33"}) {
+								age
+								active
+								name
+								points
+								created
+								metadata
+								image
+							}
+						}`,
+				Results: map[string]any{
+					"create_Users": []map[string]any{
+						{
+							"age":      int64(50),
+							"active":   false,
+							"name":     "Alice",
+							"points":   float64(5),
+							"created":  time.Time(time.Date(2024, time.June, 18, 1, 0, 0, 0, time.UTC)),
+							"metadata": "{\"two\":2}",
+							"image":    "aabb33",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	testUtils.ExecuteTestCase(t, test)
+}
+
+func TestMutationCreate_WithDefaultValue_NoValueProvided_CreatedTwice_ReturnsError(t *testing.T) {
+	test := testUtils.TestCase{
+		Description: "Simple create mutation, with default value, no value provided, and created twice",
+		Actions: []any{
+			testUtils.SchemaUpdate{
+				Schema: `
+					type Users {
+						name: String @default(string: "Bob")
+						age: Int @default(int: 40)
+					}
+				`,
+			},
+			testUtils.Request{
+				Request: `mutation {
```

Comment:
question: Would it be worth using the `CreateDoc` action instead to test across all clients?

Reply:
This applies to all the integration tests.

Reply:
Yes, please use the CreateDoc action instead Keenan (nice spot Fred)

Reply:
done